js_methods.rb

```diff
@@ -0,0 +1,160 @@
+# frozen_string_literal: true
+
+# rubocop:disable Style/StringLiterals, Style/AndOr
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield to_a[i] if block_given?
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield to_a[i], i if block_given?
+      i += 1
+    end
+  end
+
+  def my_select
+    temp_arr = []
+    my_each do |e|
+      temp = yield e
+      temp_arr << e if temp
+    end
+    temp_arr
+  end
+
+  def my_all?
+    ans = true
+    my_each do |e|
+      ans = false unless yield e
+    end
+    ans
+  end
+
+  def my_any?
+    ans = false
+    my_each do |e|
+      ans = true if yield e
+    end
+    ans
+  end
+
+  def my_none?
+    ans = true
+    my_each do |e|
+      ans = false if yield e
+    end
+    ans
+  end
+
+  def my_count(*chr)
+    # why can't use *char directly???
+    cnt = 0
+    char = *chr
+    if !char.empty?
+      my_each { |e| cnt += 1 if e == char[0] }
+      cnt
+    else
+      length
+    end
+  end
```

Comment:
**[REQUIRED]**
* Counts the number of elements yielding a true value if a block is given